app/controllers/middleware_server_controller.rb

```diff
@@ -324,27 +350,28 @@ def run_server_operation(operation_info, items)
           run_operation_on_record(operation_info, mw_server)
         end
         operation_triggered = true
+        log_operation_in_timeline(operation_info, mw_server)
       end
     end
     add_flash(_("%{operation} initiated for selected server(s)") % {:operation => operation_info.fetch(:msg)}) if operation_triggered
   end
 
   def trigger_mw_operation(operation, mw_server, params = nil)
     mw_manager = mw_server.ext_management_system
-    path = mw_server.ems_ref
-
-    # in domain mode case we want to run the operation on the server-config DMR resource
-    if mw_server.respond_to?(:in_domain?) && mw_server.in_domain?
-      path = path.sub(/%2Fserver%3D/, '%2Fserver-config%3D')
-    end
 
-    args = [operation, path]
+    args = [operation, mw_server.ems_ref]
     if mw_server.instance_of?(MiddlewareDeployment)
       args << mw_server.name
     elsif params
       args << params
     end
 
+    # in domain mode case we want to run the operation on the server-config DMR resource
+    if mw_server.respond_to?(:in_domain?) && mw_server.in_domain?
```

Comment:
Wouldn't it be better to place this `if` couple lines above, so you're not relying on `args[1]` magic and you're rather adding already substituted path? I mean something like:

```
args = [operation]
if mw_server.respond_to?(:in_domain?) && mw_server.in_domain?
    args << mw_server.ems_ref.sub(/%2Fserver%3D/, '%2Fserver-config%3D')
    args << {:original_resource_path => mw_server.ems_ref }
else
    args << mw_server.ems_ref
end
```
Because if anything would be added into the array any patch later, the path might not longer be on index 1.

Reply:
I was just trying to avoid testing twice if `mw_server` is a domain server, because I need `{:original_resource_path => mw_server.ems_ref }` in the last position.

Hmm... So, if the elements in the array are re-ordered, it would mean that whatever method is called by the following `mw_manager.public_send(*args)` in line 375 may have changed it's signature, which is a breaking change anyway. So, not sure if I really should the mw_server test twice.

Reply:
Maybe you can do something like `*[operation, path].push(*args)` to avoid using `args[1]`

Reply:
I think it's better to just skip the array and pass the positional parameters explicitly :)